fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -345,6 +304,41 @@ def query_categories(category_keywords: Dict[str, List[str]] = KEYWORDS) -> pd.D
     return df_raw
 
 
+# TODO: need to account for spelling variations w/in a new colum in the google sheet
+# todo: $validate-code: there are actually 2 different operations w/ this same name. might need to disambiguate
+def _get_keywords(
+    sheet_name='category_keywords', cache_path=CONFIG['keywords_cache_path'], use_cache_only=False
+) -> Dict[str, List[str]]:
+    """Get keywords from google sheets, else cache
+    sheet_name: The name of the specific sheet within a GoogleSheet workbook."""
+    # Load
+    if use_cache_only:
+        df: pd.DataFrame = pd.read_csv(cache_path)
+    else:
+        try:
+            # todo: Pass URI too
+            df: pd.DataFrame = get_sheets_data(sheet_name=sheet_name, env_dir=ENV_DIR)
+            df.to_csv(cache_path, index=False)
+        except:
+            df: pd.DataFrame = pd.read_csv(cache_path)
+
+    # Convert to dictionary
+    category_keywords: Dict[str, List[str]] = {}
+    categories = list(df['category'].unique())
+    for c in categories:
+        df_i = df[df['category'] == c]
+        keywords = list(df_i['keyword'])
+        category_keywords[c] = keywords
+
+    return category_keywords
+
+
+def run():
```

Comment:
This is the initializing function now. Gets keywords first and then passes them.